### ISSUE TITLE
Lost ability to reference ID'ed component in xml

### ISSUE THREAD

**Issue by hoseyjoe** [haxeui/haxeui-core#339]
Basically
```xml
<vbox>
   <box id="home" />
</vbox>
```
var main = ComponentMacros.createInstance("assets/ui/main.xml");
main.home through an errror.  

The fix is
```
 haxe/ui/macros/ModuleMacros.hx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/haxe/ui/macros/ModuleMacros.hx b/haxe/ui/macros/ModuleMacros.hx
index ac590b5..6be5029 100644
--- a/haxe/ui/macros/ModuleMacros.hx
+++ b/haxe/ui/macros/ModuleMacros.hx
@@ -357,7 +357,7 @@ class ModuleMacros {
             pack: superClassParts
         }
         var xml = sys.io.File.getContent(fullPath);
-        var namedComponents:Map<String, String> = new Map<String, String>();
+        var namedComponents:Map<String, ComponentMacros.NamedComponentDescription> = new Map<String, ComponentMacros.NamedComponentDescription>();
         var codeBuilder = new CodeBuilder();
         ComponentMacros.buildComponentFromString(codeBuilder, xml, namedComponents);
         codeBuilder.add(macro {
@@ -392,7 +392,7 @@ class ModuleMacros {
         var classBuilder = new ClassBuilder(newClass.fields, Context.currentPos());
         
         for (name in namedComponents.keys()) {
-            var typeClass = namedComponents.get(name);
+            var typeClass = namedComponents.get(name).type;
             var typeParts = typeClass.split(".");
             var typeName = typeParts.pop();
             var t:TypePath = {
```

**Comment by ianharrigan** [comment]
Im not totally sure i understand the problem / fix... can you create a PR, even if its for reference?

Cheers,
Ian

**Comment by hoseyjoe** [comment]
I went to an old project and compile it and I got
```
src/Main.hx:27: characters 9-13 : assets.ui.Main has no field home
```
the XML was like
```xml
<vbox>
   <box id="home" />
</vbox>
```

The problem is there were no namedComponents because the buildComponentFromString was passing in a different map than that which was expected

Map<String, String> versus Map<String, NamedComponentDescription> 

In buildComponentFromString namedComponents has a value but when it came back it was null.


Anyways.  Never done a PR. I think I need to branch then commit to my branch then create PR?  I don't think I can make PR from local repo can I
